lib/ansiblelint/generate_docs.py

```diff
@@ -60,15 +58,10 @@ def import_all_rules():
 
 
 def get_serialized_rules():
-    mod_list = [m[1] for m in getmembers(rules) if ismodule(m[1])]
-    class_list = []
-    for mod in mod_list:
-        class_temp = [m[1] for m in getmembers(mod) if isclass(m[1])]
-        class_temp = [c for c in class_temp if c is not AnsibleLintRule]
-        class_list.extend(class_temp)
+    # called after all classes are loaded
 
     all_rules = []
-    for c in class_list:
+    for c in AnsibleLintRule.__subclasses__():
```

Comment:
I'm not sure if this'd work because you'd have to import all the modules in this runtime first.

Reply:
So I'd leave this change out of the PR.

Reply:
This is `what import_all_rules` does just before calling `get_serialized_rules`. The proof that it works is that the docs are not affected by this change.

I do have many questions regarding how the  generate_docs.py was originally implemented but that would be subject of other changes. For the moment I did only what was needed in order to allow me to move AnsibleLintRule definition inside.